certificates.go

```diff
@@ -2,13 +2,21 @@ package gomatrixserverlib
 
 import (
 	"crypto/x509"
+	"errors"
 )
 
 // IsValidCertificate checks if the given x509 certificate can be verified using
 // system root CAs and an optional pool of intermediate CAs.
 func IsValidCertificate(serverName ServerName, c *x509.Certificate, intermediates *x509.CertPool) (valid bool, err error) {
+	host, _, isValid := ParseAndValidateServerName(serverName)
+	if !isValid {
+		return false, errors.New("Not a valid serverName")
```

Comment:
I'm planning to have these errors appear in the output JSON and thus be human-readable. Can we print out which serverName it's validating in this error for easier debugging?

Reply:
done